replica/table.cc

```diff
@@ -1544,7 +1545,60 @@ class static_compaction_group_manager final : public compaction_group_manager {
     }
 };
 
+class tablet_compaction_group_manager final : public compaction_group_manager {
+    replica::table& _t;
+private:
+    const locator::effective_replication_map_ptr& erm() const {
+        return _t.get_effective_replication_map();
+    }
+
+    const schema_ptr& schema() const {
+        return _t.schema();
+    }
+
+    const locator::tablet_map& tablet_map() const {
+        // FIXME: cheaper way to retrieve tablet_map than looking up every time in tablet_metadata's map.
+        auto& tm = erm()->get_token_metadata();
+        return tm.tablets().get_tablet_map(schema()->id());
+    }
+public:
+    tablet_compaction_group_manager(replica::table& t)
+            : _t(t) {
+    }
+
+    std::vector<std::unique_ptr<compaction_group>> make_compaction_groups() const override {
+        std::vector<std::unique_ptr<compaction_group>> ret;
+        auto& tmap = tablet_map();
+        auto& tm = erm()->get_token_metadata();
+        ret.reserve(tmap.tablet_count());
+
+        for (auto tid : tmap.tablet_ids()) {
+            auto group_id = fmt::format("{}/{}", tid, tmap.tablet_count());
```

Comment:
Could we generate group id on demand to save on memory, and computation here?

Reply:
it was introduced by dabf46c37ff4f5b0dd2b7acddc4e8d12604dc3e4

I think it's even more correct to generate it on demand as # of tablets can change. we only need each group to know its id.